### PR TITLE
Generate `profiles.yml` on the fly

### PR DESCRIPTION
**Pull request by swtwsk**
Generates `profiles.yml` on the fly, based on `config/{env}/dbt.yml` and `config/{env}/{target_type}.yml` data

- Depends on #3 
- Requires acceptance of `target_type` in `DbtExecutionEnvironmentParameters` constructor in [dbt-airflow-manifest-parser](https://github.com/getindata/dbt-airflow-manifest-parser)

---
Keep in mind:
- [ ] Documentation updates
- [X] [Changelog](CHANGELOG.md) updates

### CODE REVIEW

File: data_pipelines_cli/cli_commands/compile.py [hunk 1]
```diff
@@ -62,6 +63,10 @@ def _docker_build(docker_args: DockerArgs):
 
 
 def _dbt_compile(env: str) -> None:
+    echo_info("Generating .profiles.yml")
+    profiles_path = generate_profiles_yml(env)
+    echo_subinfo(f"Generated .profiles.yml in {profiles_path}")
```
[the remark this answers is not in the document]
Comment (p-pekala) [reply data_pipelines_cli/cli_commands/compile.py:68]: The same below, then you don't need to use spaces between methods :p

File: data_pipelines_cli/config_generation.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    target_type_config = read_config(
+        BUILD_DIR.joinpath("dag"), env, f"{dbt_target_type}.yml"
+    )
+
+    profile = {
+        dbt_target_type: {
+            "target": dbt_target,
+            "outputs": {dbt_target: target_type_config},
+        }
+    }
```
Comment (p-pekala) [inline data_pipelines_cli/config_generation.py:23]: I would extract it to function called: generate_profile_json

File: setup.py [hunk 3]
```diff
@@ -13,6 +13,7 @@
     "copier==5.1.0",
     "dbt>=0.21, <0.22",
     "fsspec",
+    "dbt-airflow-manifest-parser>=0.14",
```
Comment (p-pekala) [inline setup.py:16]: chyba bezpieczniej jest fixować wersję

File: tests/dag/config/base/bigquery.yml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+type: bigquery
```
Comment (p-pekala) [inline tests/dag/config/base/bigquery.yml:1]: I think we can skip the type  and take always from dbt.yml (you already have target_type)

File: tests/dag/config/snow/snowflake.yml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+type: snow
```
Comment (p-pekala) [inline tests/dag/config/snow/snowflake.yml:1]: same as for bq

**Review by p-pekala — CHANGES_REQUESTED** [review]

File: setup.py (hunk 3 above)
Comment (mariusz89016) [reply setup.py:16]: nb: a przynajmniej do minora (choć `0.x.y` nie zakłada wstecznej kompatybilności to chyba fajnie ją trzymać chociażby dla bugfixów)

File: tests/dag/config/base/bigquery.yml [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+method: service-account
+project: example-project
+dataset: example-dataset
+threads: 1
+keyfile: /var/keyfile.json
+timeout_seconds: 150
+priority: interactive
+location: us-west1
+retries: 1
```
Comment (mariusz89016) [inline tests/dag/config/base/bigquery.yml:10]: nb: newline (is there formatter/linter in this project?)

File: tests/test_config_generation.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+                self.current_dir_path.joinpath(
+                    "example_profiles", f"{env}_{profile_type}.yml"
+                ),
+                "r",
+            ) as prepared:
+                self.assertDictEqual(
+                    yaml.safe_load(generated), yaml.safe_load(prepared)
+                )
+
+    def test_dev_bigquery_generation(self):
```
Comment (mariusz89016) [inline tests/test_config_generation.py:44]: nb: maybe subtests? https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests

File: tests/dag/config/base/bigquery.yml (hunk 6 above)
Comment (swtwsk) [reply tests/dag/config/base/bigquery.yml:10]: We have linter and formatter for Python (`isort`, `black` and `flake8`), but not for YAML files. Apart from that, those yamls are just for test purposes.

File: tests/test_config_generation.py (hunk 7 above)
Comment (swtwsk) [reply tests/test_config_generation.py]: changed to subtests

File: tests/dag/config/base/bigquery.yml (hunk 4 above)
Comment (swtwsk) [reply tests/dag/config/base/bigquery.yml]: done

File: data_pipelines_cli/config_generation.py (hunk 2 above)
Comment (swtwsk) [reply data_pipelines_cli/config_generation.py:36]: extracted

**Review by p-pekala — APPROVED** [review]

**Pull request merged**